### PR TITLE
[W5.4][W14-A1][Teo Wen Zong]

### PR DESCRIPTION
**Pull request by wenzongteo**
Add SortCommand.java
Add PrepareSort in Parser.java
Add Test Case in ParserTest.java
Add Sort by asc or dsc
Update input.txt & expected.txt
Update Help command
Update User Guide
Update Test Cases

### CODE REVIEW

**Comment by HouDenghao** [comment]
Suggestions: you can list automatically after executing the sort command so every time you do not need to call list command. It may save a lot of time.

**Comment by wenzongteo** [comment]
Thank you for your idea Deng Hao!

**Comment by chuaweiwen** [comment]
Nice! Didn't realise you can capture the extra arguments (i.e. asc, dsc) like this.

**Comment by wenzongteo** [comment]
Updated the sort feature with @HouDenghao 's suggestion!

File: src/seedu/addressbook/commands/Command.java [hunk 1]
```diff
@@ -37,6 +37,16 @@ public static String getMessageForPersonListShownSummary(List<? extends ReadOnly
     }
 
     /**
+     * Constructs a feedback message to summarise an operation that displayed a listing of persons after the user sorts.
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/Command.java]: The comment is too wordy :stuck_out_tongue:  consider making it concise.

File: src/seedu/addressbook/commands/SortCommand.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.person.UniquePersonList;
+
+import java.util.List;
+
+/**
+ * Sort records in the AddressBook
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/SortCommand.java]: *Sorts* records...

File: src/seedu/addressbook/data/person/Person.java [hunk 3]
```diff
@@ -34,6 +35,30 @@ public Person(ReadOnlyPerson source) {
         this(source.getName(), source.getPhone(), source.getEmail(), source.getAddress(), source.getTags());
     }
 
+    /**
+     * Comparator to sort people by their name in ascending order.
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Person.java]: a better header comment would be `compares people by their name for sorting in ascending order`

File: src/seedu/addressbook/data/person/Person.java [hunk 4]
```diff
@@ -34,6 +35,30 @@ public Person(ReadOnlyPerson source) {
...
+     */
+    public static Comparator<Person> sortByNameASC = new Comparator<Person>(){
+        public int compare(Person p1, Person p2){
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Person.java]: need space between `) {`

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 5]
```diff
@@ -1,11 +1,6 @@
 package seedu.addressbook.data.person;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.Iterator;
-import java.util.List;
+import java.util.*;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/UniquePersonList.java]: Don't import using `*` import the classes explicitly, only what is necessary

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 6]
```diff
@@ -123,6 +118,19 @@ public void remove(ReadOnlyPerson toRemove) throws PersonNotFoundException {
     }
 
     /**
+     * Sort all persons in list by name. 1 for ascending and 0 for descending
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/UniquePersonList.java]: your options are not the same as the ones given in the header comments.
please write appropriate header comments.

File: test/input.txt [hunk 7]
```diff
@@ -23,7 +23,7 @@
   list
 
 ##########################################################
-# test add person command, setup state for futuer tests
+# test add person command, setup state for future tests
```
Comment (okkhoy) [inline test/input.txt]: :+1:

**Comment by okkhoy** [comment]
@wenzongteo 
Some comments added. Please ack & close the PR after reading comments.

File: src/seedu/addressbook/data/person/Person.java (hunk 3 above)
Comment (wenzongteo) [reply src/seedu/addressbook/data/person/Person.java]: Should we also include pre and post conditions in our comments?

Comment (wenzongteo) [reply src/seedu/addressbook/data/person/Person.java]: @okkhoy
